manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -79,24 +80,25 @@
 
     @Transactional
     @Override
-    public Processor getProcessor(String processorId, String bridgeId, String customerId) {
-
-        Bridge bridge = bridgesService.getBridge(bridgeId, customerId);
+    public Processor getProcessor(String processorId, Bridge bridge) {
         Processor processor = processorDAO.findByIdBridgeIdAndCustomerId(processorId, bridge.getId(), bridge.getCustomerId());
         if (processor == null) {
-            throw new ItemNotFoundException(String.format("Processor with id '%s' does not exist on Bridge '%s' for customer '%s'", processorId, bridgeId, customerId));
+            throw new ItemNotFoundException(String.format("Processor with id '%s' does not exist on Bridge '%s' for customer '%s'", processorId, bridge.getId(), bridge.getCustomerId()));
         }
 
         return processor;
     }
 
     @Override
-    public Processor createProcessor(String bridgeId, String customerId, ProcessorRequest processorRequest) {
+    public Processor createProcessor(Bridge bridge, ProcessorRequest processorRequest) {
         /* We cannot deploy Processors to a Bridge that is not Available */
-        Bridge bridge = bridgesService.getReadyBridge(bridgeId, customerId);
+        if (!bridgesService.isBridgeReady(bridge)) {
```

Comment:
This doesn't feel right. Determining whether or not a Bridge is ready should be a concern of the `BridgesService`. The previous implementation was correct IMHO with the `BridgesService.getReadyBridge` returning either the ready bridge instance or throwing the `BridgeLifecycleException`.

Reply:
done